freud/density.pyx

```diff
@@ -576,10 +576,17 @@ cdef class RDF(_SpatialHistogram1D):
     behavior of :math:`\lim_{r \to \infty} g(r)=1` is recovered. However, for
     very small systems the long range behavior of the radial distribution will
     instead tend to :math:`\frac{N-1}{N}`. In small systems, where this
-    deviation is noticeable, the ``normalize`` flag may be used to rescale the
-    results and force the long range behavior to 1. Note that this option will
-    have little to no effect on larger systems (for example, for systems of 100
-    particles the RDF will differ by 1%).
+    deviation is noticeable, the ``normalization_mode`` argument may be used to
+    rescale the results and force the long range behavior to 1. Note that this
+    option will have little to no effect on larger systems (for example, for
+    systems of 100 particles the RDF will differ by 1%).
+
+    .. note::
+        The ``normalization_mode`` argument should not be used if
+        :code:`query_points` is provided as a different set of points, or if
+        unusual query arguments are provided to :meth:`~.compute`, specifically
+        if :code:`exclude_ii` is set to :code:`False`. This normalization is
+        not meaningful in such cases and will simply convolute the data.
```

Comment:
This would be a complex set of conditions to check. For a typical use case of the RDF module, `compute` will be called repeatedly on many frames in a trajectory in order to accumulate a smooth RDF curve. Checking the conditions in the code would create a lot of often unnecessary overhead, so we choose to document these cases instead of checking them explicitly for the performance benefit.